src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/MachObjectWriter.cs

```diff
@@ -351,6 +351,9 @@ private protected override void CreateSection(ObjectNodeSection section, string
                 }
             };
 
+            if (section.Name != "hydrated" && section.Name != "bss")
+                flags |= S_ATTR_NO_DEAD_STRIP;
```

Comment:
I suppose this can be just appended to the `flags` initialization above:

```
            uint flags = section.Name switch
            {
                "bss" => S_ZEROFILL,
                ".eh_frame" => S_COALESCED,
                _ => section.Type == SectionType.Uninitialized ? S_ZEROFILL : (S_REGULAR | S_ATTR_NO_DEAD_STRIP)
            };
```

Reply:
Yes, thanks. The change was just to run some tests + seems like failures might be related.

Reply:
I don't think it matters for that section since it contains unwind information and not code/data. It has special processing in the linker.